imsim/treerings.py

```diff
@@ -22,14 +22,14 @@ class TreeRings():
     _req_params = {'file_name' : str}
     _opt_params = {'only_dets' : list}
 
-    def __init__(self, file_name, only_dets=None, logger=None):
+    def __init__(self, file_name, only_dets=None, logger=None, defer_load=True):
         """
         Constructor.
         Craig Lage UC Davis 19-Mar-18; cslage@ucdavis.edu
         This code reads in a file with tree ring parameters from file_name
         and assigns a tree ring model to each sensor.
         """
-        logger = galsim.config.LoggerWrapper(logger)
+        self.logger = galsim.config.LoggerWrapper(logger)
```

Comment:
This is likely to cause problems, since loggers aren't picklable.  Better to pass the logger to functions that need it each time, rather than save it as an attribute.